.github/workflows/ci.yml

```diff
@@ -94,3 +94,39 @@ jobs:
         directory: ./build
         files: ./build/coverage.out
         fail_ci_if_error: true
+
+  format:
+    name: Formatting on ${{ matrix.platform.name }}
+    runs-on: ${{ matrix.platform.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+        - { name: Linux, os: ubuntu-latest }
+        - { name: macOS, os: macos-12 }
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v2
+
+    - name: Install Linux Dependencies
+      if: runner.os == 'Linux'
+      run: sudo apt-get update && sudo apt-get install clang-format-12
+
+    - name: Install macOS Dependencies
+      if: runner.os == 'macOS'
+      run: brew install clang-format
+
+    - name: Format Code on Linux
+      if: runner.os == 'Linux'
+      run: cmake -DCLANG_FORMAT_EXECUTABLE=clang-format-12 -P $GITHUB_WORKSPACE/cmake/Format.cmake
+
+    - name: Format Code on macOS
+      if: runner.os == 'macOS'
+      run: cmake -DCLANG_FORMAT_EXECUTABLE=/usr/local/Cellar/clang-format/14.0.5/bin/clang-format -P $GITHUB_WORKSPACE/cmake/Format.cmake
```

Comment:
Should this be as hard coded? Wouldn't this break if a new patch or minor version is released?
Or asked differently, if we hard code it like that, shouldn't we also ensure that the specific version is installed via brew?

Reply:
> Should this be as hard coded? 

Probably not.

> Wouldn't this break if a new patch or minor version is released?

Probably yes.

I never have to do this on my own macOS machine so surely there's a better way to get that clang-format binary into our `PATH`. Because Brew installs to different paths on x86 and ARM, I don't have a great way of debugging this but hopefully an x86 Mac user can help me out. I'd expect that binary to also exist in `/user/local/bin/clang-format` but I haven't tried that path yet.

Reply:
I figured out what I was doing wrong. I changed `-DCLANG_FORMAT_EXECUTABLE=/usr/local/Cellar/clang-format/14.0.5/bin/clang-format` to `-DCLANG_FORMAT_EXECUTABLE=clang-format` which is possible because that executable is symlinked into /usr/loca/bin which is already part of the `PATH`.